lib/timers.js

```diff
@@ -580,13 +580,15 @@ function processImmediate() {
   while (immediate) {
     domain = immediate.domain;
 
-    if (!immediate._onImmediate)
-      continue;
```

Comment:
@hassy Could you try re-running the tests on your machine with the following patch applied?

When I tried moving the conditional above, it ran fine locally.

```diff
diff --git a/lib/timers.js b/lib/timers.js
index 37ac317..d89a1d8 100644
--- a/lib/timers.js
+++ b/lib/timers.js
@@ -578,17 +578,18 @@ function processImmediate() {
   immediateQueue.head = immediateQueue.tail = null;
 
   while (immediate) {
+
+    if (!immediate._onImmediate) {
+      immediate = immediate._idleNext;
+      continue;
+    }
+
     domain = immediate.domain;
 
     if (domain)
       domain.enter();
 
-    if (immediate._onImmediate) {
-      immediate._callback = immediate._onImmediate;
-    } else {
-      immediate = immediate._idleNext;
-      continue;
-    }
+    immediate._callback = immediate._onImmediate;
 
     // Save next in case `clearImmediate(immediate)` is called from callback
     var next = immediate._idleNext;
```